### DISCUSSION TITLE
Get session or accessToken inside a Helper class

### DISCUSSION THREAD

**Discussion by quangas** [nextauthjs/next-auth#2102] [Help] [answered]
## Question 💬

Once you have the session, you can utilize the hooks useSession and getSession on various pages. I have a helper class which has all my API endpoints, and I need to get the accessToken from useSession (session object) but within the helper class I cannot use the hook. Is there some way to retrieve the session within the helper class without passing it in?

Api.js

```
export default {
  createTodo: async function(data) {
    const res = await fetch(`${API_URL}/todos`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json",
        "Authorization": `${this.get_access_token_from_session()}`
      }
    });
    const json = await res.json();

    return json;
  },
}
```

Yes, I can do `createTodo(access_token_from_page, data)` but I wanted the `function get_access_token_from_session()` in the Api.js class to be able to somehow read the cookie/session and just get the token for me.

Any ideas or is there some better approach?

**Comment by balazsorban44** [comment, marked as answer]
You can get the session by doing a fetch call to : `/api/auth/session`

https://next-auth.js.org/getting-started/rest-api#get-apiauthsession

alternatively you can proxy your API requests through a Next.js API route and use either `getSession` or `getToken` and attach the access token there before sending the request to your external api.

**Reply by quangas** [reply]
Amazing, thanks so much. The fetch call to /api/auth/session does what I need.

Do you have any links for the second route? Just curious to see this alternative.

**Reply by balazsorban44** [reply]
I meant https://nextjs.org/docs/api-routes/introduction

together with https://next-auth.js.org/getting-started/client#getsession

or

https://next-auth.js.org/tutorials/securing-pages-and-api-routes#using-gettoken

depending on if you use JWT or DB sessions.
